api/cloud/oc_cloud_manager.c

```diff
@@ -554,7 +580,7 @@ on_keepalive_response_default(oc_cloud_context_t *ctx, bool response_received,
     }
     ++ctx->retry_count;
   }
-  return !is_retry_over(ctx);
+  return !(ctx->retry_count >= OC_ARRAY_SIZE(g_retry_timeout));
```

Comment:
can be simplified to `ctx->retry_count < OC_ARRAY_SIZE(g_retry_timeout)`

Reply:
fixed